plugins/woocommerce-admin/client/guided-tours/variable-product-tour/index.tsx

```diff
@@ -0,0 +1,131 @@
+import { __ } from '@wordpress/i18n';
+import { TourKit, TourKitTypes } from '@woocommerce/components';
+import { useUserPreferences } from '@woocommerce/data';
+import { recordEvent } from '@woocommerce/tracks';
+
+function getStepName(
+	steps: TourKitTypes.WooStep[],
+	currentStepIndex: number
+) {
+	return steps[ currentStepIndex ]?.meta?.name;
+}
+
+export const VariableProductTour = () => {
+	const [ isTourOpen, setIsTourOpen ] = useState( false );
+
+	const { updateUserPreferences, variable_product_tour_shown: hasShownTour } =
+		useUserPreferences();
+
+	const config: TourKitTypes.WooConfig = {
+		steps: [
+			{
+				referenceElements: {
+					desktop: '.attribute_tab',
+				},
+				focusElement: {
+					desktop: '.attribute_tab',
+				},
+				meta: {
+					name: 'attributes',
+					heading: __( 'Start by adding attributes', 'woocommerce' ),
+					descriptions: {
+						desktop: __(
+							'Add attributes like size and color for customers to choose from on the product page. We will use them to generate product variations.',
+							'woocommerce'
+						),
+					},
+					primaryButton: {
+						text: __( 'Got it', 'woocommerce' ),
+					},
+				},
+			},
+		],
+		options: {
+			callbacks: {
+				onNextStep: ( currentStepIndex ) => {
+					recordEvent( 'variable_product_tour_step_viewed', {
+						step: getStepName( config.steps, currentStepIndex + 1 ),
+					} );
+				},
+			},
+			// WooTourKit does not handle merging of default options properly,
+			// so we need to duplicate the effects options here.
+			effects: {
+				spotlight: {
+					interactivity: {
+						enabled: true,
+						rootElementSelector: '#wpwrap',
+					},
+				},
+				arrowIndicator: true,
+				liveResize: {
+					mutation: true,
+					resize: true,
+					rootElementSelector: '#wpwrap',
+				},
+			},
+		},
+		closeHandler: ( steps, currentStepIndex ) => {
+			updateUserPreferences( { variable_product_tour_shown: 'yes' } );
+			setIsTourOpen( false );
+
+			if ( currentStepIndex === steps.length - 1 ) {
+				recordEvent( 'variable_product_tour_completed', {
+					step: getStepName(
+						steps as TourKitTypes.WooStep[],
+						currentStepIndex
+					),
+				} );
+			} else {
+				recordEvent( 'variable_product_tour_dismissed', {
+					step: getStepName(
+						steps as TourKitTypes.WooStep[],
+						currentStepIndex
+					),
+				} );
+			}
+		},
+	};
+
+	// show the tour when the product type is changed to variable
+	useEffect( () => {
+		const productTypeSelect = document.querySelector(
+			'#product-type'
+		) as HTMLSelectElement;
+
+		if ( hasShownTour === 'yes' || ! productTypeSelect ) {
+			return;
+		}
+
+		function handleProductTypeChange() {
+			if ( productTypeSelect.value === 'variable' ) {
+				setIsTourOpen( true );
+				recordEvent( 'variable_product_tour_started', {
+					step: getStepName( config.steps, 0 ),
+				} );
+				recordEvent( 'variable_product_tour_step_viewed', {
+					step: getStepName( config.steps, 0 ),
+				} );
```

Comment:
I think that recording this event now is a little redundant. We can remove it, and in the future, if we decide to add more steps to the spotlight, bring this back.

Reply:
Is there harm in including this now?

Reply:
I like having these in initially for consistency with other tour Tracks and so that if we decide to add additional steps in the future we don't need to go back and do this... it just works. (Side note: it would be great to make this more general and allow us to set up Tracks for tours in a generic manner so we didn't need to implement it for each tour).

Reply:
> Is there harm in including this now?

I feel that adding three events for this specific spotlight is a little overkill. In my opinion, distinguishing between steps will make sense only when we have more than one. Especially considering that adding Tracks events is easy, but removing them is harder.

I would wait until we use this spotlight as a tour to add the Tracks event related to `steps`.

> it would be great to make this more general and allow us to set up Tracks for tours in a generic manner so we didn't need to implement it for each tour

+1 on that! Maybe we can create an issue to tackle that in the future.

Reply:
I can remove these Tracks events.

Reply:
Removed in e2e2875